polkadot-parachains/parachains-common/src/impls.rs

```diff
@@ -98,14 +98,33 @@ where
 	}
 }
 
+/// Tests `loc` whether it starts with `prefix`.
+/// Similar to `MultiLocation::match_and_split` but allows arbitrary suffixes (instead of just one
+/// junction).
+// TODO: Replace with version implemented on MultiLocation here:
+// https://github.com/paritytech/polkadot/pull/4827
```

Comment:
XCM v3 might take a while to be merged. I can just create an issue to track this.